portability/memory.cc

```diff
@@ -270,7 +270,7 @@ toku_realloc(void *p, size_t size) {
 }
 
 void *toku_realloc_aligned(size_t alignment, void *p, size_t size) {
-#if __APPLE__
+#if defined(__APPLE__) && __APPLE__
```

Comment:
just if defined or ifdef, no need to check the actual value of the macro, according to the docs (https://developer.apple.com/library/content/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html#//apple_ref/doc/uid/TP40002850-SW13)